catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -1053,6 +1225,37 @@ public QueryResponse query(QueryRequest queryRequest, FederationStrategy strateg
                         queryRequest.getProperties());
             }
 
+            if (policy != null) {
+                HashMap<String, Set<String>> requestPolicyMap = new HashMap<>();
+                Map<String, Serializable> unmodifiableProperties = Collections.unmodifiableMap(
+                        queryReq.getProperties());
+                for (PolicyPlugin plugin : policy) {
+                    try {
+                        PolicyResponse policyResponse = plugin.processPreQuery(queryReq.getQuery(), unmodifiableProperties);
+                        buildPolicyMap(requestPolicyMap, policyResponse.operationPolicy().entrySet());
+                    } catch (PluginExecutionException e) {
+                        LOGGER.info(
+                                "Plugin processing failed. This is allowable. Skipping to next plugin.",
+                                e);
+                    } catch (StopProcessingException e) {
```

Comment:
The query handles the stop processing exception in a slightly different way than the other operations. It still fails the operation.